### PR TITLE
unsafeDepositTx

### PR DESCRIPTION
**Pull request by wilsoncusack**

### CODE REVIEW

File: _test/utils.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  details: string
+
+  constructor(code: number, message: string) {
+    super(message)
+    this.code = code
+    this.details = message
+  }
+}
+
+export const anvilChain = {
```
Comment (roninjin10) [inline _test/utils.ts:27]: doesn't anvil exist in @wagmi/chains?

File: src/actions/wallet/unsafeDepositTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { OpChainL2 } from '@roninjin10/rollup-chains'
+import { writeContract } from 'viem/actions'
+
+type DepositTransactionParameters = {
+  to: Address
+  value: bigint
+  gasLimit: bigint
+  isCreation: boolean
+  data: Hex
+}
```
Comment (roninjin10) [inline src/actions/wallet/unsafeDepositTransaction.ts:22]: Later we should not hardcode this 

```typescript
type DepositTransactionParameters = AbiParametersToPrimitiveTypes<
  ExtractAbiFunction<typeof optimismPortalABI, 'depositTransaction'>['inputs']
 >
```

**Review by roninjin10 — APPROVED** [review]
lgtm

File: _test/utils.ts (hunk 1 above)
Comment (wilsoncusack) [reply _test/utils.ts:27]: I don't think it does 🤔

**Pull request merged**